psi4/src/psi4/libmints/matrix.cc

```diff
@@ -317,8 +317,8 @@ void Matrix::copy(const Matrix *cp) {
 #pragma omp parallel for
     for (int h = 0; h < nirrep_; ++h) {
         if (rowspi_[h] != 0 && colspi_[h ^ symmetry_] != 0)
-            memcpy(&(matrix_[h][0][0]), &(cp->matrix_[h][0][0]),
-                   rowspi_[h] * (size_t)colspi_[h ^ symmetry_] * sizeof(double));
+            memmove(&(matrix_[h][0][0]), &(cp->matrix_[h][0][0]),
```

Comment:
the two blocks of memory are guaranteed to be non-overlapping, right?

Reply:
According to valgrind they're not. This is from the cherry-picked commit message:
```
==31415== Source and destination overlap in memcpy(0x1e6851c0,0x1e6851c0, 1800)
==31415==    at 0x4A0A403: memcpy@@GLIBC_2.14 (in /usr/lib64/valgrind/vgpreload_memcheck-amd64-linux.so)
==31415==    by 0x11E100E: psi::Matrix::copy(psi::Matrix const*) [clone ._omp_fn.0] (in build/bin/psi4)
==31415==    by 0x11E311A: psi::Matrix::copy(psi::Matrix const*) (in build/bin/psi4)
==31415==    by 0x1208D8D: psi::Wavefunction::load_orbitals() (in build/bin/psi4)
==31415==    by 0x9B240D: psi::scf::HF::guess() (in build/bin/psi4)
==31415==    by 0x9B68F7: psi::scf::HF::compute_energy() (in build/bin/psi4)
==31415==    by 0x82BBAF: psi::scf::scf(psi::Options&, _object*, object*) (in build/bin/psi4)
```

Reply:
Ah, they have the same address.  Weird.  It's trying to copy a matrix to itself somewhere.  I think your fix is correct and addresses the symptom, but we should probably also try and address the cause in another PR - a good candidate for Psicon.